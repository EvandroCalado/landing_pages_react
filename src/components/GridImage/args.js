export const args = {
  background: false,
  title: 'My grid',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.',
  grid: [
    {
      altText: 'Teste 1',
      image:'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      altText: 'Teste 2',
      image:'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      altText: 'Teste 3',
      image:'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      altText: 'Teste 4',
      image:'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      altText: 'Teste 5',
      image:'https://source.unsplash.com/random/800x800?r=5',
    },
    {
      altText: 'Teste 6',
      image:'https://source.unsplash.com/random/800x800?r=6',
    },
  ],
};
