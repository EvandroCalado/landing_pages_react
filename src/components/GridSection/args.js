export const args = {
  background: false,
  title: 'My grid',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.',
    },
  ],
};
