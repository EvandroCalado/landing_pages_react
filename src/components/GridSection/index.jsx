import P from 'prop-types';
import * as Styled from './styles';
import SectionBackgound from '../SectionBackground';
import Heading from '../Heading';
import TextComponent from '../TextComponent';

const GridSection = ({
  title,
  description,
  grid,
  sectionId,
  background = false,
}) => {
  return (
    <SectionBackgound background={background} sectionId={sectionId}>
      <Styled.Container>
        <Heading size="huge" uppercase colorDark={!background} as="h2">
          {title}
        </Heading>
        <TextComponent>{description}</TextComponent>
        <Styled.Grid>
          {grid.map((item) => (
            <Styled.GridElement key={item.title}>
              <Heading size="medium" colorDark={!background} as="h3">
                {item.title}
              </Heading>
              <TextComponent>{item.description}</TextComponent>
            </Styled.GridElement>
          ))}
        </Styled.Grid>
      </Styled.Container>
    </SectionBackgound>
  );
};

GridSection.propTypes = {
  title: P.string.isRequired,
  description: P.string.isRequired,
  background: P.bool,
  grid: P.arrayOf(
    P.shape({
      title: P.string,
      description: P.string,
    })
  ).isRequired,
  sectionId: P.string,
};

export default GridSection;
