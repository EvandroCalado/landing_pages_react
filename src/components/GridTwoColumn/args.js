export const args = {
  title: 'Grid Two Column',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Consequatur, omnis qui cumque reiciendis accusantium reprehenderit optio voluptates incidunt. Fugit facilis debitis officia repellat ab laudantium voluptatibus incidunt expedita a nihil!`,
  image: "assets/images/javascript.svg"
}