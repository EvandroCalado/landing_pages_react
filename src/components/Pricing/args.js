export const args = {
  id: 1,
  __component: 'section.pricing',
  description_one:
    'The release of Apple Silicon-based Macs at the end of last year generated a flurry of news coverage and some surprises at the machine’s performance. This post details some background information on the experience of porting Firefox to run natively on these CPUs.',
  description_two:
    'We’ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.',
  title: 'pricing',
  table: [
    {
      id: 1,
      head: 'title 1',
      body: 'content1',
      footer: 'footer 1',
    },
    {
      id: 2,
      head: 'title 2',
      body: 'content2',
      footer: 'footer 2',
    },
    {
      id: 3,
      head: 'title 3',
      body: 'content3',
      footer: 'footer 3',
    },
    {
      id: 4,
      head: 'title 4',
      body: 'content4',
      footer: 'footer 4',
    },
    {
      id: 5,
      head: 'title 5',
      body: 'content5',
      footer: 'footer 5',
    },
  ],
  matadata: {
    id: 7,
    name: 'pricing',
    section_id: '#pricing',
    background: false,
  },
};
