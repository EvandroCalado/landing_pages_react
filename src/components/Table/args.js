export const args = [
  {
    id: 1,
    head: "title 1",
    body: "content 1",
    footer: "footer 1"
  },
  {
    id: 2,
    head: "title 2",
    body: "content 2",
    footer: "footer 2"
  },
  {
    id: 3,
    head: "title 3",
    body: "content 3",
    footer: "footer 3"
  },
  {
    id: 4,
    head: "title 4",
    body: "content 4",
    footer: "footer 4"
  },
  {
    id: 5,
    head: "title 5",
    body: "content 5",
    footer: "footer 5"
  }
]