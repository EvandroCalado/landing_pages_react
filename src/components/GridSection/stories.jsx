import GridSection from '.';
import { args } from './args';

const GridSectionStories = {
  title: 'GridSection',
  component: GridSection,
  args: args,
};

export const Template = (args) => {
  return (
    <div>
      <GridSection {...args} />
    </div>
  );
};

export default GridSectionStories;
