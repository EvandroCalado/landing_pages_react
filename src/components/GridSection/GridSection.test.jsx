import GridSection from '.';
import { renderTheme } from '../../styles/render.theme';

import { args } from './args';

describe('<GridSection />', () => {
  it('should render with background', () => {
    const { container } = renderTheme(<GridSection {...args} />);
    expect(container).toMatchSnapshot();
  });

  it('should render without background', () => {
    const { container } = renderTheme(<GridSection {...args} background={undefined} />);
    expect(container).toMatchSnapshot();
  });
});
