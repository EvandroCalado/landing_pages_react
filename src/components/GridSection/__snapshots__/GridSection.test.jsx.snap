// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridSection /> should render with background 1`] = `
.c4 {
  color: #0a1128;
  font-size: 6.4rem;
  font-weight: 900;
  text-transform: uppercase;
  text-align: center;
}

.c8 {
  color: #0a1128;
  font-size: 3.2rem;
  font-weight: 900;
  text-transform: none;
  text-align: center;
}

.c10 {
  font-size: 2.4rem;
}

.c2 .c9 {
  margin-bottom: 6.4rem;
}

.c5 {
  margin-bottom: 6.4rem;
  text-align: center;
}

.c6 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
  overflow: hidden;
  width: 100%;
}

.c7 .c3 {
  position: relative;
  left: 5rem;
  text-align: left;
}

.c7 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3.5rem;
  left: -5rem;
  -webkit-transform: rotate(15deg);
  -ms-transform: rotate(15deg);
  transform: rotate(15deg);
}

.c1 {
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #fff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          My grid
        </h2>
        <h2
          class="c5"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.
        </h2>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <h3
              class="c3 c8"
            >
              Teste 1
            </h3>
            <div
              class="c9 c10"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.
            </div>
          </div>
          <div
            class="c7"
          >
            <h3
              class="c3 c8"
            >
              Teste 2
            </h3>
            <div
              class="c9 c10"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.
            </div>
          </div>
          <div
            class="c7"
          >
            <h3
              class="c3 c8"
            >
              Teste 3
            </h3>
            <div
              class="c9 c10"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridSection /> should render without background 1`] = `
.c4 {
  color: #0a1128;
  font-size: 6.4rem;
  font-weight: 900;
  text-transform: uppercase;
  text-align: center;
}

.c8 {
  color: #0a1128;
  font-size: 3.2rem;
  font-weight: 900;
  text-transform: none;
  text-align: center;
}

.c10 {
  font-size: 2.4rem;
}

.c2 .c9 {
  margin-bottom: 6.4rem;
}

.c5 {
  margin-bottom: 6.4rem;
  text-align: center;
}

.c6 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
  overflow: hidden;
  width: 100%;
}

.c7 .c3 {
  position: relative;
  left: 5rem;
  text-align: left;
}

.c7 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3.5rem;
  left: -5rem;
  -webkit-transform: rotate(15deg);
  -ms-transform: rotate(15deg);
  transform: rotate(15deg);
}

.c1 {
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #fff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          My grid
        </h2>
        <h2
          class="c5"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.
        </h2>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <h3
              class="c3 c8"
            >
              Teste 1
            </h3>
            <div
              class="c9 c10"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.
            </div>
          </div>
          <div
            class="c7"
          >
            <h3
              class="c3 c8"
            >
              Teste 2
            </h3>
            <div
              class="c9 c10"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.
            </div>
          </div>
          <div
            class="c7"
          >
            <h3
              class="c3 c8"
            >
              Teste 3
            </h3>
            <div
              class="c9 c10"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
